### ISSUE TITLE
NPE when reading from Pubsub

### ISSUE THREAD

**Issue by nevillelyh** [spotify/scio#480]
Code:
```scala
object PubWriteTest {
  def main(cmdlineArgs: Array[String]): Unit = {
    val (sc, args) = ContextAndArgs(cmdlineArgs)
    sc.parallelize((1 to 100).map("message-" + _))
      .saveAsPubsub("projects/scio-playground/topics/neville-test")
    sc.close()
  }
}

object PubReadTest {
  def main(cmdlineArgs: Array[String]): Unit = {
    val (sc, args) = ContextAndArgs(cmdlineArgs)
    sc.pubsubTopic[String]("projects/scio-playground/topics/neville-test")
      .debug()
    sc.close().waitUntilFinish()
  }
}
```

```
Exception in thread "main" java.lang.NullPointerException
	at org.apache.beam.sdk.io.PubsubUnboundedSource$PubsubReader.ackBatch(PubsubUnboundedSource.java:640)
	at org.apache.beam.sdk.io.PubsubUnboundedSource$PubsubCheckpoint.finalizeCheckpoint(PubsubUnboundedSource.java:313)
	at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.getReader(UnboundedReadEvaluatorFactory.java:174)
	at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.processElement(UnboundedReadEvaluatorFactory.java:127)
	at org.apache.beam.runners.direct.TransformExecutor.processElements(TransformExecutor.java:139)
	at org.apache.beam.runners.direct.TransformExecutor.run(TransformExecutor.java:107)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by nevillelyh** [comment]
Might be related to this: http://stackoverflow.com/questions/41465231/pubsubreader-failed-with-nullpointerexception-when-executed-with-inprocesspipeli

**Comment by ravwojdyla** [comment]
This seems like a different problem. Coming from pubsub client being null at:

```
org.apache.beam.sdk.io.PubsubUnboundedSource$PubsubReader.ackBatch(PubsubUnboundedSource.java:639)
  at org.apache.beam.sdk.io.PubsubUnboundedSource$PubsubCheckpoint.finalizeCheckpoint(PubsubUnboundedSource.java:312)
  at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.finishRead(UnboundedReadEvaluatorFactory.java:223)
  at org.apache.beam.runners.direct.UnboundedReadEvaluatorFactory$UnboundedReadEvaluator.processElement(UnboundedReadEvaluatorFactory.java:144)
  at org.apache.beam.runners.direct.TransformExecutor.processElements(TransformExecutor.java:139)
  at org.apache.beam.runners.direct.TransformExecutor.run(TransformExecutor.java:107)
  at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
  at java.util.concurrent.FutureTask.run(FutureTask.java:266)
  at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
  at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
  at java.lang.Thread.run(Thread.java:745)
```

Also tried the most recent 0.7.0-SNAPSHOT of beam, and the problem is still reproducable. Will apply patch locally, and submit upstream.

**Comment by ravwojdyla** [comment]
beam PR -> apache/beam#2365

**Comment by ravwojdyla** [comment]
closing - now solved in `0.7.0`, makes no sense to monkey patch this one.
